finality-aleph/src/party/mod.rs

```diff
@@ -432,40 +459,32 @@ where
 
     /// Returns authorities for a given session or None if the first block of this session is
     /// higher than the highest finalized block
-    fn get_authorities_for_session(&mut self, session_id: SessionId) -> Option<Vec<AuthorityId>> {
-        let mut session_authorities = self.session_authorities.lock();
-        if let Some(authorities) = session_authorities.get(&session_id) {
-            Some(authorities.clone())
-        } else {
-            let authorities = if session_id == SessionId(0) {
-                self.client
-                    .runtime_api()
-                    .authorities(&BlockId::Number(<NumberFor<B>>::saturated_from(0u32)))
-                    .unwrap()
-            } else {
-                let last_finalized_number = self.client.info().finalized_number;
-                let first_block = first_block_of_session::<B>(session_id, self.session_period);
-                if last_finalized_number >= first_block {
-                    return None;
-                }
-
-                match self.client.runtime_api().next_session_authorities(&BlockId::Number(first_block)) {
-                    Ok(authorities) => authorities
-                        .expect("Authorities for next session must be available at first block of current session"),
-                    Err(e) => {
-                        error!(target: "afa", "Error when getting authorities for session {:?} {:?}", session_id, e);
-                        panic!("We didn't get the authorities for the session {:?}", session_id);
-                    }
-                }
-            };
-            session_authorities.insert(session_id, authorities.clone());
-            Some(authorities.clone())
+    fn updated_authorities_for_session(
+        &mut self,
+        session_id: SessionId,
+    ) -> Option<Vec<AuthorityId>> {
+        let last_finalized_number = self.client.info().finalized_number;
+        let first_block = first_block_of_session::<B>(session_id, self.session_period);
+        if last_finalized_number < first_block {
+            return None;
         }
+
+        Some(
+            self.session_authorities
+                .lock()
```

Comment:
Shouldn't this be the `await`y lock?

Reply:
https://github.com/Cardinal-Cryptography/aleph-node/pull/276/files#diff-1abd0b64c271c22b933a2edd8813eaa74ef220d1ff5d8b8f1c36ccf2689e1fdeL436

This line suggesting that there shouldn't be, but if you think differently I will change it

Reply:
Done

Reply:
Link doesn't point to a line. :/ Or just doesn't work.